WI #634: Commitment Discount Flexibility Appendix (#535)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/appendix/commitment_discounts/size_flexibility.md

> # Size Flexibility

Reviewer: Side note: If we call this "size flexibility", I believe this will lock us into `SkuSize` for the thing that conveys the actual size/shape/SKU/edition/etc of the resource.

> A usage-based commitment discount obligates a customer to a usage amount for one or more related SKUs in return for reduced rates.  For example, when a customer purchases a usage-based commitment discount to cover a specific database SKU, this commitment will cover every hour over the term where at least one instance of this SKU is running. It can also cover eligible resources wholly, where 1 hour is covered by 1 resource, or partially, where 1 committed hour can be split across multiple resources or 1 resource can be fully covered with unused commitment discount costs. These latter two examples are commonly known as *size-flexibility*.

Reviewer: Minor nits...
```suggestion
A usage-based commitment discount obligates a customer to a usage amount for one or more related SKUs in return for reduced rates.  For example, when an organization purchases a usage-based commitment discount to cover a specific database SKU, this commitment will cover every hour over the term where at least one instance of this SKU is running. It can also cover eligible resources wholly, where 1 hour is covered by 1 resource, or partially, where 1 committed hour can be split across multiple resources or 1 resource can be fully covered with unused commitment discount costs. These latter two examples are commonly known as *size flexibility*.
```

> Since providers have different rules for when size-flexibility is or is not enabled, the following, made-up SKU pricing attributes will be used in each example below, and each example will be categorized under size-flexibility enabled or disabled sections.

Reviewer: ```suggestion
Since providers have different rules for when size flexibility is or is not enabled, the following, fictitious SKU pricing attributes will be used in each example below, and each example will be categorized under size flexibility enabled or disabled sections.
```

## SKUs & Rates

> | ProviderName | Service | SkuId     | ListUnitPrice  | CommittedUnitPrice | NormalizationFactor |

Reviewer: I know you're making up these extra columns, but shouldn't we at least follow our naming patterns?
```suggestion
| ProviderName | Service | SkuId     | ListUnitPrice  | x_EffectiveUnitPrice | x_NormalizationFactor |
```

| ------------ | ------- | --------- | -------------- | -------------------| ------------------- |
| TinyCloud    | Compute | VM_Small  | $0.50          | $0.25              | 1                   |
| TinyCloud    | Compute | VM_Medium | $1.00          | $0.50              | 2                   |
| TinyCloud    | Compute | VM_Large  | $2.00          | $1.00              | 4                   |
| TinyCloud    | Compute | VM_XLarge | $4.00          | $2.00              | 8                   |

> The above, made-up SKU pricing attributes show that this provider, *TinyCloud*, only has 1 service that offers 4 virtual machine SKUs with various list rates, committed rates, and normalization factors. Each SKU's normalization factor classifies its relative size to its committed rate. Usage-based commitment discounts with size-flexibility can fully cover any combination of 1 or more SKUs where the sum of their normalization factor equals the normalization factor of the commitment discount.

Reviewer: ```suggestion
The above, fictitious SKU pricing attributes show that this provider, *TinyCloud*, only has 1 service that offers 4 virtual machine SKUs with various list rates, committed rates, and normalization factors. Each SKU's normalization factor classifies its relative size to its committed rate. Usage-based commitment discounts with size flexibility can fully cover any combination of 1 or more SKUs where the sum of their normalization factor equals the normalization factor of the commitment discount.
```

> ### Scenarios *without* size-flexibility

Reviewer: ```suggestion
### Scenarios *without* size flexibility
```

        "ChargePeriodEndDate": "2023-01-01T01:00:00Z",
        "ChargeCategory": "Usage",
        "ChargeFrequency": "Usage-Based",
        "PricingCategory": "Committed",
        "ResourceId": "<my-large-vm-id>",
        "PricingQuantity": 1,
        "BilledCost": 0.00,
        "EffectiveCost": 1.00,
        "ConsumedUnit": 1,
        "ConsumedUnit": "Hour",
        "SkuId": "VM_Large",
        "CommitmentDiscountId": "<commitment-discount-id>",
        "CommitmentDiscountStatus": "Unused",
        "CommitmentDiscountQuantity": 1,
        "CommitmentDiscountUnit": "Hour"
    },
    {
        "BillingPeriodStartDate": "2023-01-01T00:00:00Z",
        "BillingPeriodEndDate": "2023-02-01T00:00:00Z",
        "ChargePeriodStartDate": "2023-01-01T00:00:00Z",
        "ChargePeriodEndDate": "2023-01-01T01:00:00Z",
        "ChargeCategory": "Usage",
        "ChargeFrequency": "Usage-Based",
        "PricingCategory": "Standard",
        "ResourceId": "<my-medium-vm-id>",
        "PricingQuantity": 1,
        "BilledCost": 1.00,
        "EffectiveCost": 0.00,
        "ConsumedUnit": 1,
        "ConsumedUnit": "Hour",
        "SkuId": "VM_Medium"
    }
]
```

> ### Scenarios *with* size-flexibility

Reviewer: ```suggestion
### Scenarios *with* size flexibility
```
